kratos/sources/model_part.cpp

```diff
@@ -192,7 +192,10 @@ void ModelPart::Reset()
     // construct a new variable list and process info. Old data ptrs is not destroyed
     // since, same data may be shared with some other model parts as well.
     mpVariablesList = Kratos::make_intrusive<VariablesList>();
-    mpProcessInfo = Kratos::make_shared<ProcessInfo>();
+    // only reset mpProcessInfo if this is not a sub model part because sub model parts have nullptr
+    if (!IsSubModelPart()) {
+        mpProcessInfo = Kratos::make_shared<ProcessInfo>();
+    }
```

Comment:
take care, here you are creating a new one in the root, but you also need to assign it to the submodelparts...

Reply:
Do you mean by re-adding [these lines](https://github.com/KratosMultiphysics/Kratos/pull/13376#discussion_r2085347502) that were removed before?
```suggestion
 else {
        mpProcessInfo = nullptr;
 }
```

Reply:
well, possibly i am wrong since this should never happen. please discard my comment

Reply:
correct me if I'm wrong @RiccardoRossi, but this is related to this previous comment on @sunethwarna's review, right?
https://github.com/KratosMultiphysics/Kratos/pull/13376#discussion_r2086143910